MSAL/src/MSALPublicClientApplication.m

```diff
@@ -827,10 +827,12 @@ - (void)acquireTokenSilentWithParameters:(MSALSilentTokenParameters *)parameters
     /*
      In the acquire token silent call we assume developer wants to get access token for account's home tenant,
      if authority is a common, organizations or consumers authority.
+     TODO: update instanceAware parameter to the instanceAware in config
      */
     NSError *authorityError = nil;
     requestAuthority = [self.msalOauth2Provider issuerAuthorityWithAccount:parameters.account
                                                           requestAuthority:requestAuthority
+                                                             instanceAware:NO
```

Comment:
Do we want to keep this very specific for instance aware or have some generic interface that can accept multiple custom parameters? This likely depends if we're anticipating any other parameters like instance aware to change the output authority.

Reply:
I think we could accept the config object here as parameters, what do you think?

Reply:
Config object is MSAL thing and this goes to common core, so it wouldn't be present, right?

Reply:
I think the config object can be pass all the way down until here:
https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/613/files#diff-bacbc810f667825154d2fd12fcdc17d1L104

where the `MSIDAccountMetadataCacheAccessor` will take all individual parameters that are needed to store/retrieve cache url.

Or you think it is better to construct `MSIDRequestParameters` and then pass along in the above call at line 833?

Reply:
discussed offline with Olga. I have changed the account metadata schema such that it could be easily extended in future and minimize the risk of breaking change.